### PR TITLE
[Alignment/HIPAlignmentAlgorithm] modernize `LhcTrackAnalyzer`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all.

#### PR validation:

Privately run `Alignment/HIPAlignmentAlgorithm/test/LHCAnalyzer_cfg.py`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/HIPAlignmentAlgorithm/plugins/LhcTrackAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+  const auto& vtx = vertices.front();
+  if (vtx.isFake()) {
+    goodvtx_ = false;
+  } else {
+    goodvtx_ = true;
+  }
+
+  goodbx_ = true;
+  // int bx = iEvent.bunchCrossing();
+  //if (bx==51 || bx==2724) goodbx_=true;
```
Comment (tvami) [inline Alignment/HIPAlignmentAlgorithm/plugins/LhcTrackAnalyzer.cc:169]: Hi @mmusich do we need these commented-out lines? Btw is `goodbx_` ever set to not-true? I didn't see  anything in the code that would do that, so to me it seems this variable is not used. Is that right?

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+alca

**Comment by qliphy** [comment]
+1

**Comment by iarspider** [comment]
@mmusich it seems that this update broke IBs: [gcc10](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc10/CMSSW_12_1_X_2021-08-16-2300/Alignment/HIPAlignmentAlgorithm), [clang](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc900/CMSSW_12_1_CLANG_X_2021-08-16-2300/Alignment/HIPAlignmentAlgorithm). Also QA now complains about duplicate dictionaries: [link](https://cmssdt.cern.ch/SDT/cgi-bin/showDupDict.py/slc7_amd64_gcc900/www/mon/12.1-mon-23/CMSSW_12_1_X_2021-08-16-2300/testLogs/dupDict-edmPD.log)

**Comment by mmusich** [comment]
@iarspider see https://github.com/cms-sw/cmssw/pull/34903

**Comment by iarspider** [comment]
Ah. Sorry!

**Pull request merged**